source/features/highest-rated-comment.tsx

```diff
@@ -0,0 +1,125 @@
+import React from 'dom-chef';
+import select from 'select-dom';
+import features from '../libs/features';
+import './highest-rated-comment.css';
+
+type Option = {
+	$el: Element;
+	likes: number;
+	unlikes: number;
+	index: number;
+}
+type Props = {
+	id: string;
+	username: string;
+	text: string;
+	avatar: string;
+}
+
+const like = '👍';
+const unlike = '👎';
+
+const element = ({id, username, text, avatar}: Props): Node => (
+	<div className="timeline-comment-wrapper rgh-highest-rated-comment">
+		<div className="avatar-parent-child timeline-comment-avatar">
+			<a href={`https://github.com/${username}`} className="d-inline-block">
+				<img src={avatar} alt={username} className="avatar rounded-1" height="44" width="44"/>
+			</a>
+		</div>
+
+		<a href={`#${id}`}>
+			<div className="bg-white border details-reset rounded-1">
+				<div className="bg-gray border-bottom-0 px-2 py-0 rgh-highest-rated-comment-summary">
+					<div className="d-flex flex-items-center">
+						<a href={`#${id}`} className="btn btn-sm rgh-highest-rated-comment-btn">
+							<svg height="16" className="octicon octicon-arrow-down" viewBox="0 0 10 16" version="1.1" width="20" aria-hidden="true">
+								<path fill-rule="evenodd" d="M7 7V3H3v4H0l5 6 5-6H7z"></path>
+							</svg>
+						</a>
+
+						<div className="text-gray timeline-comment-header-text rgh-highest-rated-comment-text">
+							<span>Highest-Rated Comment: {text}</span>
+						</div>
+					</div>
+				</div>
+			</div>
+		</a>
+	</div>
+);
+
+function init(): void {
```

Comment:
In this function you can simplify things a lot, but you'll have to fully rewrite it.

Select the comments that have 👍 reactions, so we skip a lot of them already. You can do so by selecting the specific button, and use a simple `for-of` to find the highest non-controversial comment:


```js
function getCount(reaction): number {
	return Number(reaction.textContent!.match(/\d+/)![0]!);
}

let highest;
for (const like of select.all('.reaction-summary-item[value="THUMBS_UP react"]')) {
	const count = getCount(like);
	const dislike = select('[value="THUMBS_DOWN react"]', like.parentElement);
	if (dislike && getCount(dislike) >= count / 2) {
		continue; // Controversial comment
	}
	if (!highest) {
		highest = {like, count};
	} else if (count > highest.count) {
		highest = {like, count};
	}
}
```

at the end of it, you'll have exactly the highest-rated non-controversial comment. Then just make sure its likes are enough (>10, > comments * 0.8) and add it.

```js
if (highest && highest.count > 10) {
	append(element(highest.like.closest('.comment')))
}
```

Reply:
I'm on this right now. Found out `THUMBS_UP react` becomes `THUMBS_UP unreact` upon reaction so I'm using `.reaction-summary-item[value="THUMBS_UP react"],.reaction-summary-item[value="THUMBS_UP unreact"]` if that's okay.

Reply:
Should I still verify if it's at least the 5th comment?

Reply:
We can do that later in the selector. I’ll send a patch

Reply:
> I'm on this right now. Found out `THUMBS_UP react` becomes `THUMBS_UP unreact` upon reaction so I'm using `.reaction-summary-item[value="THUMBS_UP react"],.reaction-summary-item[value="THUMBS_UP unreact"]` if that's okay.

Try `[aria-label*="reacted with thumbs up"]` and `[aria-label*="reacted with thumbs down"]` for the dislike button

Reply:
Works just fine.

Reply:
So this would be the selector: `.js-discussion .js-timeline-item:nth-child(n+6) [aria-label*="reacted with thumbs up"]`

n+6 because there's a hidden item just after the first comment so to get the actual 5th elemente we need +6.

What do you think?

Reply:
Yes! The only issue is that `.js-timeline-item` includes labels and title change events, so it's not completely reliable, but maybe it's "good enough" for now.